test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -447,6 +456,1909 @@ public async Task GetInstalledVersion_WithAssetsFile_ChangingPackageSpec_Returns
+                        var project = CreateTestNetCorePackageReferenceProject(projectName, projectFullPath, projectCache);
+
+                        // We need to treat NU1605 warning as error.
+                        project.IsNu1605Error = true;
+                        netCorePackageReferenceProjects.Add(project);
+                        testSolutionManager.NuGetProjects.Add(project);
+
+                        //Let new project pickup my custom package source.
+                        project.ProjectLocalSources.AddRange(sources);
+                        var projectNames = GetTestProjectNames(projectFullPath, projectName);
+                        var packageSpec =  GetPackageSpec(
+                            projectName,
+                            projectFullPath,
+                            packageA_Version100.Version);
+
+                        if (prevPackageSpec != null)
+                        {
+                            packageSpec = packageSpec.WithTestProjectReference(prevPackageSpec);
+                        }
+
+                        // Restore info
+                        var projectRestoreInfo = ProjectJsonTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                        projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                        projectCache.AddProject(projectNames, projectAdapter, project).Should().BeTrue();
+                        prevProj = projectFullPath;
+                        packageSpecs[i] = packageSpec;
+                        prevPackageSpec = packageSpec;
+                        projectFullPaths[i] = projectFullPath;
+                    }
+
+                    var initialInstalledPackages = await netCorePackageReferenceProjects[numberOfProjects-1].GetInstalledPackagesAsync(CancellationToken.None);
+
+                    // Act
+                    var results = await nuGetPackageManager.PreviewProjectsInstallPackageAsync(
+                        netCorePackageReferenceProjects, // All projects
+                        packageB,
+                        resolutionContext,
+                        testNuGetProjectContext,
+                        sourceRepositoryProvider.GetRepositories(),
+                        CancellationToken.None);
+
+                    var actions = results.Select(a => a.Action).ToArray();
+
+                    await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
+                        netCorePackageReferenceProjects,
+                        actions,
+                        testNuGetProjectContext,
+                        new SourceCacheContext(),
+                        CancellationToken.None);
+
+                    // Assert
+                    var builtIntegratedActions = actions.OfType<BuildIntegratedProjectAction>().ToList();
+                    Assert.Equal(actions.Count(), builtIntegratedActions.Count());
```

Comment:
Please assert that the expected package is actually installed. 
For all we know, this could've just restored the current project state :)

Reply:
Added assertion for action package Id and package version.